src/app/products/products-mix/view-product-mix/view-product-mix.component.ts

```diff
@@ -0,0 +1,50 @@
+/** Angular Imports */
+import { Component, OnInit, ViewChild } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { MatPaginator, MatSort, MatTableDataSource } from '@angular/material';
+
+/**
+ * View product mix component.
+ */
+@Component({
+  selector: 'mifosx-view-product-mix',
+  templateUrl: './view-product-mix.component.html',
+  styleUrls: ['./view-product-mix.component.scss']
+})
+export class ViewProductMixComponent implements OnInit {
+
+  /** Product mix data. */
+  productMixData: any;
+  /** Allowed products datasource. */
+  allowedProductsDatasource: MatTableDataSource<any>;
+  /** Restricted products datasource. */
+  restrictedProductsDatasource: MatTableDataSource<any>;
+  /** Columns to be displayed in allowed products table. */
+  allowedProductsDisplayedColumns: string[] = ['name'];
+  /** Columns to be displayed in restricted products table. */
+  restrictedProductsDisplayedColumns: string[] = ['name'];
+
+  /** Paginator for allowed products table. */
+  @ViewChild('allowed') allowedPaginator: MatPaginator;
+  /** Paginator for restricted products table. */
+  @ViewChild('restricted') restrictedPaginator: MatPaginator;
+  /** Sorter for allowed products table. */
+  @ViewChild(MatSort) allowedSort: MatSort;
+  /** Sorter for restricted products table. */
+  @ViewChild(MatSort) restrictedSort: MatSort;
+
+  constructor(private route: ActivatedRoute) {
+    this.route.data.subscribe((data: { productMix: any }) => {
+      this.productMixData = data.productMix;
+    });
+  }
+
+  ngOnInit() {
+    this.allowedProductsDatasource = new MatTableDataSource(this.productMixData.allowedProducts);
+    this.restrictedProductsDatasource = new MatTableDataSource(this.productMixData.restrictedProducts);
+    this.allowedProductsDatasource.paginator = this.allowedPaginator;
+    this.restrictedProductsDatasource.paginator = this.restrictedPaginator;
+    this.allowedProductsDatasource.sort = this.allowedSort;
+    this.restrictedProductsDatasource.sort = this.restrictedSort;
+  }

```

Comment:
kindly shift all allowedProduct statements to setAllowedProducts() function, and the same for restricted products.